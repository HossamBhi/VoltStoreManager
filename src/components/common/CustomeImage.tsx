import React, {useState} from 'react';
import {Image, ImageProps, StyleSheet} from 'react-native';

export default ({source, style, resizeMode, ...props}: ImageProps) => {
  const [imageUrl, setImageUrl] = useState(source);

  return (
    <Image
      testID="image"
      source={imageUrl}
      style={[styles.image, style]}
      resizeMode={resizeMode}
      onError={() => setImageUrl(require('../../assets/logo.png'))}
      {...props}
    />
  );
};

const styles = StyleSheet.create({
  image: {minWidth: 30, minHeight: 30},
});
