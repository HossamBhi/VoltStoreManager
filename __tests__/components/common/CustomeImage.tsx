import React from 'react';
import {act, create} from 'react-test-renderer';
import CustomeImage from '../../../src/components/common/CustomeImage';

it('Render Custome Image', () => {
  const image = create(<CustomeImage />);
  const testId = image.root.findByProps({testID: 'image'}).props;
  act(() => testId.onError());
  // expect(testId.source).toBe(require('../../assets/logo.png'));
  expect(image.toJSON()).toMatchSnapshot();
});
