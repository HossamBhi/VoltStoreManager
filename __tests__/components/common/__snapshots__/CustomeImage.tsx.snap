// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Custome Image 1`] = `
<Image
  onError={[Function]}
  source={
    Object {
      "testUri": "../../../src/assets/logo.png",
    }
  }
  style={
    Array [
      Object {
        "minHeight": 30,
        "minWidth": 30,
      },
      undefined,
    ]
  }
  testID="image"
/>
`;
